tests/vmi_multus_test.go

```diff
@@ -641,18 +640,21 @@ var _ = Describe("[Serial]SRIOV", func() {
 	})
 
 	Context("VirtualMachineInstance with sriov plugin interface", func() {
-
-		getSriovVmi := func(networks []string, cloudInitNetworkData string) *v1.VirtualMachineInstance {
-
+		getSriovVmi := func(networks []string, cloudInitNetworkData string, mac ...string) *v1.VirtualMachineInstance {
```

Comment:
I do not understand why you add the `mac`.

Reply:
Before this, when we wanted to add the mac to the vm object  we would assume that the SRIOV NIC is in index 1 in the interfaces array of the vmi object, and just add it there. I think it is better to be general here and add the mac to the interface we recognize as SRIOV.  So I am adding the mac to the sriov interface when it is being defined in getSriovVmi.

Another option is to iterate over the interfaces after they are defined in the vm object, and add the mac.

Anyway I will put this in a different commit with a proper explanation.

Reply:
The problems I see here are:
- Adding another argument, requires a clear abstraction meaning to this whole function. There is little relation between a network, clould-init network-data and a mac address. It '"smells".
- Passing a list of mac/s just because you want to implement an optional argument is not a fit here. You want a mac, a single one, so you should pass a single one or generalize it to accept a list of interface{} and detect if it is the type you need (not recommended here).
- When multiple networks are used, it seems you create here multiple interfaces with the same mac, which is probably not what you intended.

The assumption that and index 1 we have SRIOV is fine here, as we know that on index 0 we place the pod network and at index 1 <= we place SRIOV interfaces. This is what this helper function does. You can use constants with names to help with this, or/and even document the helper if needed.

Unless you have a better approach here that simplifies this, I would not do this change here.

Reply:
Thank you for the detailed explanation. I understand. Will remove this change.

Reply:
> Thank you for the detailed explanation. I understand. Will remove this change.

Edit: The thing is, I still want to use `createSriovVMI` instead of using `get` ,`strat` and `wait`. 
But I would still need somehow to use an optional mac. what I can do is leave the optional mac string for 
`createSriovVMI` that is :
`createSriovVMI := func(networksNames []string, cloudInitNetworkData string, mac ...string) *v1.VirtualMachineInstance `

and wrap it in the funciton `createSriovVMIIWithStaticIPByMac` which receives a mac, not an optional mac.

so that whenever you create an sriov vmi with a costume mac, you can use `createSriovVMIIWithStaticIPByMac`, and if you create an sriov without custom mac, you can use the regular `createSriovVMI` as usual, without passing any mac.

how does that sounds (will the code also so that it'll be more concrete)

Reply:
> > Thank you for the detailed explanation. I understand. Will remove this change.
> 
> Edit: The thing is, I still want to use `createSriovVMI` instead of using `get` ,`strat` and `wait`.
> But I would still need somehow to use an optional mac. what I can do is leave the optional mac string for
> `createSriovVMI` that is :
> `createSriovVMI := func(networksNames []string, cloudInitNetworkData string, mac ...string) *v1.VirtualMachineInstance `
> 
> and wrap it in the funciton `createSriovVMIIWithStaticIPByMac` which receives a mac, not an optional mac.
> 
> so that whenever you create an sriov vmi with a costume mac, you can use `createSriovVMIIWithStaticIPByMac`, and if you create an sriov without custom mac, you can use the regular `createSriovVMI` as usual, without passing any mac.
> 
> basically just removing the optional mac from getSriovVMI.
> 
> How does that sounds (will the code also so that it'll be more concrete)

Edit 2: I can also pass to `createSriovVmi` an empty string when mac is not needed, instead of the list. and just check if the mac is empty.